packages/mdc-tab-indicator/_mixins.scss

```diff
@@ -1,60 +1,211 @@
+    }
+  }
+
+  // postcss-bem-linter: end
+}
+
+@mixin mdc-tab-indicator-surface($query: mdc-feature-all()) {
+  $feat-structure: mdc-feature-create-target($query, structure);
+
+  @include mdc-feature-targets($feat-structure) {
+    position: relative;
+  }
 }
 
-@mixin mdc-tab-indicator-underline-color($color) {
+@mixin mdc-tab-indicator-underline-color($color, $query: mdc-feature-all()) {
+  $feat-color: mdc-feature-create-target($query, color);
+
   .mdc-tab-indicator__content--underline {
-    @include mdc-theme-prop(border-color, $color);
+    @include mdc-feature-targets($feat-color) {
+      @include mdc-theme-prop(border-color, $color);
+    }
   }
 }
 
-@mixin mdc-tab-indicator-underline-height($height) {
+@mixin mdc-tab-indicator-underline-height($height, $query: mdc-feature-all()) {
+  $feat-structure: mdc-feature-create-target($query, structure);
+
   .mdc-tab-indicator__content--underline {
-    border-top-width: $height;
+    @include mdc-feature-targets($feat-structure) {
+      border-top-width: $height;
+    }
   }
 }
 
-@mixin mdc-tab-indicator-underline-top-corner-radius($radius) {
+@mixin mdc-tab-indicator-underline-top-corner-radius($radius, $query: mdc-feature-all()) {
+  $feat-structure: mdc-feature-create-target($query, structure);
+
   .mdc-tab-indicator__content--underline {
-    border-top-left-radius: $radius;
-    border-top-right-radius: $radius;
+    @include mdc-feature-targets($feat-structure) {
+      border-top-left-radius: $radius;
+      border-top-right-radius: $radius;
+    }
   }
 }
 
-@mixin mdc-tab-indicator-icon-color($color) {
+@mixin mdc-tab-indicator-icon-color($color, $query: mdc-feature-all()) {
+  $feat-color: mdc-feature-create-target($query, color);
+
   .mdc-tab-indicator__content--icon {
-    @include mdc-theme-prop(color, $color);
+    @include mdc-feature-targets($feat-color) {
+      @include mdc-theme-prop(color, $color);
+    }
   }
 }
 
-@mixin mdc-tab-indicator-icon-height($height) {
+@mixin mdc-tab-indicator-icon-height($height, $query: mdc-feature-all()) {
+  $feat-structure: mdc-feature-create-target($query, structure);
+
   .mdc-tab-indicator__content--icon {
-    height: $height;
-    font-size: $height;
+    @include mdc-feature-targets($feat-structure) {
+      height: $height;
+      font-size: $height;
+    }
   }
 }
+
+//
+// Private
+//
+
+@mixin mdc-tab-indicator-base_($query) {
+  $feat-structure: mdc-feature-create-target($query, structure);
+
+  @include mdc-tab-indicator-underline-color(primary, $query);
+  @include mdc-tab-indicator-icon-color(secondary, $query);
+  @include mdc-tab-indicator-underline-height(2px, $query);
+  @include mdc-tab-indicator-icon-height(34px, $query);
+
+  @include mdc-feature-targets($feat-structure) {
+    display: flex;
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    pointer-events: none;
+    z-index: 1;
+  }
+}
+
+@mixin mdc-tab-indicator__content_ {
+  transform-origin: left;
```

Comment:
As mentioned in my previous comment, please include feature target mixins here. This'll avoid accidentally adding future styles properties in unexpected feature target.

Same with other mixin files.

Reply:
I thought that only applied to public mixins? From what I can tell it isn't being done in any of the other private mixins, unless it involves more than one feature target.

Reply:
Can we in-line these styles above? This is essentially `mdc-tab-indicator__content_` with structure only feature target styles. I'm wary about accidentally adding a style here which is not `structure` feature target in future. 

WDYT?

Reply:
I wouldn't mind inlining them, but we did some of the other components with a similar level of granularity. This actually came in handy for the Angular Material menu since it allowed us to shave off a lot of styles that we didn't need.

Reply:
Sorry for not being clearer in my previous comment. We would still render these styles behind `structure` feature-target. I'm suggesting we can remove the need for private mixin and move it at the top.

```sass
.mdc-tab-indicator__content {
  @include mdc-feature-targets($feat-structure) {
    transform-origin: left;
    opacity: 0;
  }
}
```

instead of

```sass
.mdc-tab-indicator__content {
  @include mdc-feature-targets($feat-structure) {
    @include mdc-tab-indicator__content_;
  }
}
```

If we ever need to add additional animation related styles to `.mdc-tab-indicator__content` then we would do so by using animation feature target mixin inside the same selector.

/cc @mmalerba

Reply:
Yeah makes sense to me, it looks like some of these private mixins are just an alias for one or two properties, so you might as well just inline them. For ones that are bigger / included in multiple places we could either pass the query through or just put a comment at the top of the mixin to indicate that all of the styles belong to whichever feature